src/views/FfaListedView.vue

```diff
@@ -16,14 +16,14 @@
         v-show="selectedTab === listing"
         :ffaListing="ffaListing" />
       <button
-        v-if="enablePurchaseButton"
+        v-if="enablePurchaseButton && !canDownload"
         v-show="selectedTab === listing"
-        @click="onPurchaseClick"
+        @click="onPurchaseClicked"
         data-purchase="true">Purchase</button>
       <button
```

Comment:
If it's easy, download button should render busy while the download is happening. This depends on whether you have the state to track whether the download is in progress or completed handy.